app/components/primer/alpha/navigation_list/item.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    class NavigationList
+      # Items are rendered as styled links. They can optionally include leading and/or trailing visuals,
+      # such as icons, avatars, and counters. Items are selected based on their item_id, which is a
+      # required parameter. Items can also contain lists of subitems, which are rendered hidden but can
+      # be expanded on click.
+      class Item < Primer::Component
+        include ViewComponent::PolymorphicSlots
+
+        # The leading visual rendered before the link.
+        #
+        # @param kwargs [Hash] The arguments accepted by <%= link_to_component(Primer::Beta::Avatar) %> or <%= link_to_component(Primer::OcticonComponent) %>
```

Comment:
What about the `svg` behavior? (Perhaps this means that we should find a way to document each polymorphic slot individually?